examples/solvers/douglas_rachford_pd_tomography_tv.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Total variation sparse angle tomography using the Douglas-Rachford solver.
+
+Solves the optimization problem
+
+    min_{0 <= x <= 1, Ax = g} lam ||grad(x)||_x
+
+where ``A`` is ray transform operator, ``g`` the given noisy data, ``eps`` is
+a small number, ``grad`` is the spatial gradient and || . ||_x is the so called
+cross-norm, giving rise to isotropic Total Variation.
+
+We do this by rewriting the problem on the form
+
+    min_{0 <= x <= 1} lam ||grad(x)||_x + I_{Ax = g}
+
+where I_{.} is the indicator function, which is zero if ``Ax = g`` and infinity
+otherwise. This is a standard convex optimization problem that can
+be solved with the `douglas_rachford_pd` solver.
+
+In this example, the angles are highly under-sampled and the problem is solved
+using only 6 angles. Dispite this, we get a perfect reconstruction.
+A filtered backprojection (pseudoinverse) reconstruction is also shown at the
+end for comparsion.
+
+Note that the ``Ax = g`` condition could be relaxed to ``||Ax - g|| < eps`` for
+some small eps in order to account for noise. This would be done using the
+`IndicatorLpUnitBall` functional instead of `IndicatorZero`, as in this
+example.
+
+This is an implementation of the "puzzling numerical experiment" in the seminal
+paper "Robust Uncertainty Principles: Exact Signal Reconstruction from Highly
+Incomplete Frequency Information", Candes et. al. 2006.
+"""
+
+import numpy as np
+import odl
+
+# Parameters
+lam = 0.01
+
+# --- Create spaces, forward operator and simulated data ---
+
+# Discrete reconstruction space: discretized functions on the rectangle
+# [-20, 20]^2 with 128 samples per dimension.
+space = odl.uniform_discr(min_pt=[-20, -20], max_pt=[20, 20], shape=[512, 512])
+
+# Make a parallel beam geometry with flat detector
+# Angles: uniformly spaced, n = 22, min = 0, max = pi
+angle_partition = odl.uniform_partition(0, np.pi, 22)
```

Comment:
It's 22 in the paper, so I increased from 6 to 22 (and also increased size of domain to compensate, thats why I had the 128-512 error).